snapcraft/remote/git.py

```diff
@@ -175,12 +177,17 @@ def push_url(
             "Pushing %r to remote %r with refspec %r.", ref, stripped_url, refspec
         )
 
```

Comment:
Docstring update and a parameter are nice-to-haves but not important :
```diff
 @@ -152,6 +152,7 @@ class GitRepo:
         remote_branch: str,
         ref: str = "HEAD",
         token: Optional[str] = None,
+        push_tags: bool = False,
     ) -> None:
         """Push a reference to a branch on a remote url.

@@ -159,11 +160,15 @@ class GitRepo:
         :param remote_branch: the branch on the remote to push to
         :param ref: name of shorthand ref to push (i.e. a branch, tag, or `HEAD`)
         :param token: token in the url to hide in logs and errors
+        :param push_tags: if true, push all tags to URL (similar to `git push --tags`)

         :raises GitError: if the ref cannot be resolved or pushed
         """
        # From https://www.pygit2.org/references.html#pygit2.repository.References.iterator
        # 2 -> GIT_REFERENCES_TAGS, fetches only tags
-       tag_refs = [t.name for t in self._repo.references.iterator(2)]
+       tag_refs = [t.name for t in self._repo.references.iterator(2)] if push_tags else []
        try:
            self._repo.remotes.create_anonymous(remote_url).push([refspec] + tag_refs)
```

Reply:
this is done!